src/sentry/tsdb/base.py

```diff
@@ -169,49 +169,42 @@ def normalize_ts_to_rollup(self, epoch, seconds):
         """
         return int(epoch / seconds)
 
-    def get_optimal_rollup(self, start_timestamp, end_timestamp):
-        """
-        Identify the lowest granularity rollup available within the given time
-        range.
-        """
-        num_seconds = int(to_timestamp(end_timestamp)) - int(to_timestamp(start_timestamp))
-
-        # This loop attempts to find the smallest possible rollup that will
-        # contain both the start and end timestamps. ``self.rollups`` is
-        # ordered from the highest resolution (smallest interval) to lowest
-        # resolution (largest interval.)
-        # XXX: There is a bug here, since this function assumes that the end
-        # timestamp is always equal to or greater than the current time. If the
-        # time range is shifted far enough into the past (e.g. a 30 second
-        # window, retrieved several days after it's occurrence), this can
-        # return a rollup that has already been evicted due to TTL, even if a
-        # lower resolution representation of the range exists.
-        for rollup, samples in six.iteritems(self.rollups):
-            if rollup * samples >= num_seconds:
-                return rollup
-
-        # If nothing actually matches the requested range, just return the
-        # lowest resolution interval.
-        return list(self.rollups)[-1]
+    def get_optimal_rollup(self, start):
```

Comment:
Will need @tkaemming to chime in on this, I'm pretty clueless on TSDB.

Reply:
Yeah this was a bit of a drive-by attempt at fixing the bug mentioned in the comments and simplifying the code. but doesn't actually need to be part of this PR

Reply:
Looking back at history, I think this is OK to do (unless anything has changed since then, but I can't think of what would have): https://github.com/getsentry/sentry/pull/3860#issuecomment-238410677

I personally decided to defer it to a later change (which obviously never happened) so that we could make sure that it was working correctly without too many other changes to muddy the water in case anything went awry, your risk tolerance may differ.